curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c4.md

```diff
@@ -0,0 +1,124 @@
+---
+id: 67116d7584d0b469b14579c4
+title: Step 7
+challengeType: 1
+dashedName: step-7
+---
+
+# --description--
+
+Finally, create a `getTotalPages` function with a `catalog` parameter. The function should return the total number of all the pages of the books in the `library` array.
+
+# --hints--
+
+
+You should create a `getTotalPages` function.
+
+```js
+assert.isFunction(getTotalPages)
+```
+
+Your `getTotalPages` function should have a `catalog` parameter.
+
+```js
+assert.match(getTotalPages.toString(), /catalog/)
+```
+
+Your `getTotalPages` function should return a number.
+
+```js
+assert.isNumber(getTotalPages(library))
+```
+
+Your `getTotalPages` function should return the number `2512`.
+
+```js
+assert.equal(getTotalPages(library), 2512);
+```
+
+Your `getTotalPages` function should use an array method.
```

Comment:
same comment here as earlier. This hint will need to be more specific since we are looking for high order functions instead of any array method like push or pop